File: lastindex.cpp
#include <iostream>
using namespace std;
int rec(int a[], int size, int check)
{
    if (size == 0)
    {
        return -1;
    }
    int ans = rec(a + 1, size - 1, check);
    if (ans != -1)
    {
        return ans + 1;
    }
    if (a[0] == check)
    {
        return 0;
    }
    return -1;
}
int main()
{
    int a[6] = {1, 2, 2, 3, 4, 5};
    cout << rec(a, 6, 2);
}

File: code/lastIndex.py

def fi(a,n,x):
    if n<=0:
        return -1
    ans= fi(a,n-1,x)
    if(a[n-1]==x):
        return n-1
    return ans
a=[1,2,1,1,-7]
n=len(a)
print(fi(a,n,1))